File: code/CPP_vs_Python_list_run_time.cpp
#include <iostream>
#include <chrono> // Include the chrono library for timing

using namespace std;
using namespace std::chrono; // Use the chrono namespace for convenience

int main()
{
    // Start the timer
    auto start = high_resolution_clock::now();
    int A[1'000];
    for (int i = 0; i < 10'000; i++)
        for(int j = 0; j < 1'000; j++)
            A[j] = j;

    // Danger: Stack overflow
    // int A[10'000'000];
    // for(int j = 0; j < 10'000'000; j++)
    //      A[j] = j;

    // Stop the timer
    auto stop = high_resolution_clock::now();

    // Calculate the duration
    auto duration = duration_cast<milliseconds>(stop - start);

    // Output the time taken
    cout << "Time taken: " << duration.count() << " milliseconds" << endl;

    return 0;
}

File: code/CPP_vs_Python_list_run_time.py
import time

# Start the timer
start_time = time.time()



for i in range(10_000):
    L = []
    for j in range(1_000):
        L.append(j)

# L = []
# for j in range(10_000_000):
#     L.append(j)

# Stop the timer
end_time = time.time()

# Calculate the duration
duration = end_time - start_time

# Output the result and the time taken
# print(f"Final value of n: {n}")
print(f"Time taken: {duration:.4f} seconds")